src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size = 0;
+
+    static class Node<T> {
```

Comment:
Better to put inner class after public methods in order to have public logic before details of realization